File: ASUL/Actividades/Actividad 28-02-25-Camilo/intento.php
<?php
// Ejecutar con: php intento.php
// Fuente: https://www.php.net/manual/en/
// Version usada php 8.3.17

// Variable y ciclo
$name = "Camilo";
for ($i = 1; $i <= 3; $i++) {
    echo "Hola, $name! Intento $i\n";
}

// Condicional
$age = 21;
if ($age >= 18) {
    echo "Eres mayor de edad\n";
} else {
    echo "Eres menor de edad\n";
}

// Funcion
function saludar($nombre) {
    echo "¡Hola, $nombre!\n";
}
saludar("PHP");

// Clase y objeto
class Persona {
    public $nombre;
    function __construct($nombre) {
        $this->nombre = $nombre;
    }
    function saludar() {
        echo "¡Hola, soy $this->nombre!\n";
    }
}
$persona = new Persona("Camilo");
$persona->saludar();

// String
$name = "Camilo";

// Numero
$age = 21;

// Booleano
$isStudent = true;

echo "Nombre: $name, Edad: $age, ¿Estudiante?: " . ($isStudent ? "Si" : "No") . "\n";

// Arreglo
$frutas = ["manzana", "pera", "uva"];
echo "Primera fruta: {$frutas[0]}\n";

// Diccionario
$persona = [
    "nombre" => "Camilo",
    "edad" => 21
];
echo "Nombre: {$persona['nombre']}, Edad: {$persona['edad']}\n";

// Usar una funcion integrada
echo "Fecha actual: " . date("Y-m-d H:i:s") . "\n";

// Usar una biblioteca
use DateTime;
$fecha = new DateTime();
echo "Fecha con objeto: " . $fecha->format('Y-m-d H:i:s') . "\n";

// Tipado dinamico
$variable = "Hola";
echo "Variable: $variable\n";
$variable = 3;
echo "Variable: $variable\n";

// Debilmente tipado
$numero = "5";
$suma = $numero + 3;
echo "Suma: $suma\n";

// Puede ser fuertemente tipado
// declare(strict_types=1); // con esto al inicio del archivo

// Expresiones regulares
function es_numero($numero) {
    if (preg_match('/^[0-9]+$/', $numero)) {
        echo "$numero es un numero\n";
    } else {
        echo "$numero no es un numero\n";
    }
}
es_numero("5");
es_numero("cinco");
?>

File: ASUL/Actividades/Actividad 28-02-25-Camilo/intento.pl
#!/usr/bin/perl
# Ejecutar con: perl intento.pl
# Fuente: https://perldoc.perl.org/
# Version usada perl v5.38.3

use strict;
use warnings;

# Variable y ciclo
my $name = "Camilo";
for my $i (1..3) {
    print "Hola, $name! Intento $i\n";
}

# Condicional
if ($name eq "Camilo") {
    print "¡Hola, $name!\n";
} else {
    print "¡Hola, desconocido!\n";
}

# Funcion
sub saludar {
    my ($nombre) = @_;
    print "¡Hola, $nombre!\n";
}
saludar("Perl");

# Clases y objetos
package Persona;
sub new {
    my $class = shift;
    my $self = { nombre => shift };
    bless $self, $class;
    return $self;
}
sub saludar {
    my $self = shift;
    print "¡Hola, soy $self->{nombre}!\n";
}
my $persona = Persona->new("Camilo");
$persona->saludar();

# String
my $nombre = "Camilo";

# Numeros
my $edad = 21;

# Booleanos (Perl usa 0, undef, "" como false; cualquier otro valor es true)
my $es_estudiante = 1;

print "Nombre: $nombre, Edad: $edad, ¿Estudiante?: $es_estudiante\n";

# Arreglo
my @frutas = ("manzana", "pera", "uva");
print "Primera fruta: $frutas[0]\n";

# Hash (diccionario)
my %persona = (
    nombre => "Camilo",
    edad => 21
);
print "Nombre: $persona{nombre}, Edad: $persona{edad}\n";

# Uso de bibliotecas (modulos)
use Time::Piece;

my $time = localtime;
print "Fecha actual: $time\n";

# Tipado dinamico
my $variable = "Hola";
print "Variable: $variable\n";
$variable = 42;
print "Variable: $variable\n";

# Debilmente tipado
my $numero = "21";
my $suma = $numero + 21;
print "Suma: $suma\n";

# Expresiones regulares
sub es_numero {
    $numero = shift; # Obtener el primer parametro
    if ($numero =~ /^[0-9]+$/)
    {
        print "$numero es un numero\n";
    }
    else
    {
        print "$numero no es un numero\n";
    }
}
es_numero("42");
es_numero("hola");

